src/torrent.cpp

```diff
@@ -1121,6 +1121,12 @@ bool is_downloading_state(int const st)
 		if (m_peer_list) m_peer_list->clear();
 	}
 
+	void torrent::set_sequential_start(piece_index_t start_piece, piece_index_t end_piece)
+	{
+		if (!m_picker) return;
```

Comment:
this happens on a magnet link that hasn't received the metadata for example. I think it's important to document this behavior, that setting this range is a no-op for seeds and torrents without metadata.

I also don't think this is quite right. The piece picker is created lazily, so it's possible that we *can* create a piece picker here, we just haven't had a need for it yet.

Look at `need_picked()`, and only if its prerequisites are not satisfied should you ignore this call and return.

Reply:
Ok.

Hmm if the piece picker isn't created yet could we remember the invocation to it to ensure it is not a no-op? Like write down the range somewhere in the torrent object?

I looked at need_picker() and it seems that after checking to make sure there is valid metadata and the connections are initialized, it ensure it is not seeding (with the exception: "unless we're in suggest mode") and then it goes ahead and creates the picker.

Reply:
Think about a YouTube video URL with a start time embedded in it. In an example like that people might want to play a piece of media starting at a specific time before the torrent metadata is downloaded. In that scenario you would not want it to be a no-op, and it can be noted in a member in the torrent object, or somewhere else.